### ISSUE TITLE
seeding database

### ISSUE THREAD

**Issue by Nagacash** [medusajs/medusa#6150]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/medusajs/medusa/issues

### Issue Summary

Hey,

I have created a docker image and pasted npx create-medusa-app@latest --seed --db-url postgres://yourusername:yourpassword@localhost:5432/medusa_db into my terminal which asked me for name, email, etc. like it writing in your docs

Everything is installed until seeding database, it keeps spinning but there is no further action, it just spins. The admin panel never pops up.

How can I resolve that issue?

BR Maurice

### How can this issue be resolved?

1.
2.
3.
...

### Are you interested in working on this issue?

- [X] I would like to fix this issue

**Comment by xyzones** [comment]
@Nagacash are you sure that the db is up and running and you can connect to it with the same credentials as you are using in `create-medusa-app`

**Comment by Nagacash** [comment]
@xyzones you mean Docker Desktop yea it's running with no issue, medusa_db should be installed like create admin, etc.  
I don't get why it's not pulling the data.

**Comment by Nagacash** [comment]
✔ Created project directory
✔ Installed Next.js Starter successfully in the medusa-backend-storefront directory.
✔ Installed Dependencies
✔ Project Built
✔ Ran Migrations
✔ Created admin user

thats how far it goes

**Comment by xyzones** [comment]
@Nagacash can you connect to the DB with your credentials?

**Comment by Nagacash** [comment]
how would that work? It's coming from medusa directly

**Comment by Nagacash** [comment]
@xyzones I'm using Postgres to connect to It

**Comment by xyzones** [comment]
@Nagacash where is the postgres running at?

**Comment by Nagacash** [comment]
@xyzones DATABASE_TYPE=postgres (env) | 


version: "3.9"
services:
  db:
    image: postgres
    volumes:
      - postgres-db:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=medusa_db
      - POSTGRES_USER=yourusername
      - POSTGRES_PASSWORD=yourpassword
    ports:
      - 5432:5432
volumes:
  postgres-db:

**Comment by Nagacash** [comment]
@xyzones As you can see username and passport are both included and its getting pulled but DB is not

**Comment by xyzones** [comment]
@Nagacash I understand, but can you connect from the terminal to the db? If yes can you check if any tables have been created?

**Comment by xyzones** [comment]
@Nagacash also if the medusa is in docker as well the `DB_HOST` is not `localhost` but the postgres container name. 

in the docker-compose set the `DB_HOST`environment to the postgres container name  something like

```
  environment:
           DB_HOST: db
```

**Comment by shahednasser** [comment]
@Nagacash can you try using this snapshot with the `--verbose` option:

```bash
npx create-medusa-app@1.2.6-snapshot-20240108173057 --verbose
```

This will show you the output of every step. You can then share with us the output and where it gets stuck

**Comment by Nagacash** [comment]
@shahednasser Hey, here goes the screenshot 
![medusa](https://github.com/medusajs/medusa/assets/79047179/533c5717-5636-473a-b8bc-37bb59cf71e6)

**Comment by shahednasser** [comment]
@Nagacash can you continue till the end of the command (or until it gets stuck) then show the entire logs?

**Comment by Nagacash** [comment]
@shahednasser where do I find the Postgres username or should I just press continue?

**Comment by xyzones** [comment]
@Nagacash are you running this in docker? If yes us db as host

**Comment by Nagacash** [comment]
@shahednasser yes I'm using docker, ok let me try

**Comment by shahednasser** [comment]
> where do I find the Postgres username or should I just press continue?

You should've set this when you set up the docker postgres image. If you're unsure, you may need to delete and re-create the image to reset the password

**Comment by Nagacash** [comment]
@shahednasser 
![medusa2](https://github.com/medusajs/medusa/assets/79047179/7e311afe-89bd-49af-be12-58a5383e8e9f)
my docker image is db-1, that is my db-host, right? And username, passport as I have it inside the docker-compose file. or where would that be located?

**Comment by shahednasser** [comment]
When you first opened the issue, you said you got stuck at the "Created admin user" that means you previously provided the correct postgres username/password and medusa was able to connect to the postgres database. What username and password did you use then? Make sure that they're the same ones and that your postgres server (or docker container) is running.

**Comment by xyzones** [comment]
@Nagacash you are trying to install medusa locally or in docker?

**Comment by Nagacash** [comment]
@shahednasser I created a new docker image, it is for sure connected 
![medusa3](https://github.com/medusajs/medusa/assets/79047179/4940a697-855e-480e-bfa0-93e953706ed5)

let me run the installment once again

**Comment by Nagacash** [comment]
@shahednasser 
![medusa4](https://github.com/medusajs/medusa/assets/79047179/8a95c8fe-8cd7-4c71-9948-36004e238eec)
still the same it aint moving forward, let me try verbose once again, the problem is there is no option to set a username and passport while installing it, only once the installation is successful, the admin panel open where you can set it up

**Comment by Nagacash** [comment]
@shahednasser the data is being pulled via - POSTGRES_DB=medusa_db.
What I do not understand is why it created a user admin but did 
not pull the seeded data. Password and User are a part of the admin panel.

**Comment by shahednasser** [comment]
If you share the entire logs with the verbose options then we can figure out what the issue is.

**Comment by Nagacash** [comment]
@shahednasser I don't have a passport that's the issue
![medusa5](https://github.com/medusajs/medusa/assets/79047179/4369e870-6130-4372-93c9-b949782e002b)

**Comment by Nagacash** [comment]
@shahednasser the passport and user have to be created once I log in the first time (admin panel)

**Comment by shahednasser** [comment]
In the screenshot you provided [here](https://github.com/medusajs/medusa/issues/6150#issuecomment-1903836219) you were able to get past this step by providing the correct database user and password. Can you try again following the same steps you did then but with the `--verbose` option? 

> the passport and user have to be created once I log in the first time (admin panel)

The user/password are for the postgres database, not the admin.

**Comment by xyzones** [comment]
@Nagacash can you please start the `yt-medusa-setup` and in command line run the below

`npx create-medusa-app@latest --db-url postgresql://yourusername:yourpassword@db-1/medusa_db --with-nextjs-starter`

**Comment by Nagacash** [comment]
@shahednasser 
![medusa6](https://github.com/medusajs/medusa/assets/79047179/b3fbc107-8597-4266-ab9e-de8d333a5639)
still the same error

**Comment by xyzones** [comment]
@Nagacash db-1 as is the name of the postgres container

**Comment by shahednasser** [comment]
Can you please follow the original steps you made, with the snapshot I mentioned in [this comment](https://github.com/medusajs/medusa/issues/6150#issuecomment-1903736395) and the verbose option?

**Comment by Nagacash** [comment]
@shahednasser 
![medusa7](https://github.com/medusajs/medusa/assets/79047179/6b7f5d4d-ea81-4524-a806-d2a51c79d6e3)
the passport still remains a mystery :)

**Comment by xyzones** [comment]
> @shahednasser ![medusa2](https://private-user-images.githubusercontent.com/79047179/298564452-7e311afe-89bd-49af-be12-58a5383e8e9f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MjY0NzksIm5iZiI6MTcwNTkyNjE3OSwicGF0aCI6Ii83OTA0NzE3OS8yOTg1NjQ0NTItN2UzMTFhZmUtODliZC00OWFmLWJlMTItNThhNTM4M2U4ZTlmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDEyMjI1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkY2Y3ZTU3MTU5M2Q0ZjQ0ZDdiM2Y1M2RiZGJhMDVkZGU1NzhhYzFlZjYzZTk5Y2E5OThjYzUxNmE5MWY3NzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.zQ1yP_hfWxceptBdIe8pitv8UeV2CZlrGu8PT6VmWHs) 


here is the username and the password that you have set

**Comment by shahednasser** [comment]
@Nagacash  Unfortunately, I can't guide you with this issue unless you reach the seeding step as you did originally in the issue. Once you figure out the issue you're having with the database connection, please share the logs until you reach the seeding database step.

**Comment by Nagacash** [comment]
@shahednasser please resend the picture its says This private-user-images.githubusercontent.com page can’t be found

**Comment by Nagacash** [comment]
@shahednasser you can send me a email if possible: sonic13.ch@googlemail.com

**Comment by Nagacash** [comment]
@shahednasser thx for your effort , please send me the password via email, once I have that I can continue with the logs

**Comment by hex-grammer** [comment]
After all container running, `npm run seed` in backend exec works for me.

**Comment by sradevski** [comment]
Hey, thanks for the report! Since v2 brought a lot of architectural and API changes on the backend, we will be closing this ticket since it no longer applies to our new setup, or the issue has already been fixed. If you are still facing issues with v1,
please open a new ticket and we will address it as soon as possible. Thanks! 🙏
